base/deprecated.jl

```diff
@@ -148,8 +148,6 @@ end
 @deprecate inf{T<:AbstractFloat}(::Type{T})  convert(T,Inf)
 @deprecate nan{T<:AbstractFloat}(::Type{T})  convert(T,NaN)
 
-@deprecate_binding String AbstractString
```

Comment:
In theory @tkelman is right, but in practice I'd say it's fine in the present case, since code which wouldn't have been updated to reflect deprecations will mostly work fine, it just won't accept custom string types. Not a big deal at this point I'd say.